tests/testImageDecorrelation.py

```diff
@@ -190,65 +182,124 @@ def setUp(self):
         # Uncorrected diffim - variance is wrong (too low)
-        self.assertNotClose(np.nan_to_num(matched_im1 - im2).var(), expected_var, rtol=0.1)
-        # In fact, it should have an (incorrect) variance that is close to the variance of the science image.
-        self.assertClose((matched_im1 - im2)[~np.isnan(matched_im1 - im2)].var(), self.sig2**2., rtol=0.1)
+        self.assertNotClose(np.nan_to_num(im1 - matched_im2).var(), expected_var, rtol=0.1)
+
+        # Create the diffim (uncorrected)
+        diffExp = self.im1ex.clone()
+        tmp = diffExp.getMaskedImage()
+        tmp -= matched_im2ex.getMaskedImage()
+        # Uncorrected diffim exposure - variance is wrong (too low) - same as above but on exposure
+        statObj = afwMath.makeStatistics(diffExp.getMaskedImage(), afwMath.VARIANCECLIP,
+                                         self.statsControl)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        self.assertNotClose(var, expected_var, rtol=0.02)
 
-        diffExp = matched_im1ex.clone()
-        tmpArr = diffExp.getMaskedImage().getImage().getArray()
-        tmpArr -= self.im2ex.getMaskedImage().getImage().getArray()
-        # Uncorrected diffim exposure - variance is wrong (too low)
-        self.assertNotClose(tmpArr[~np.isnan(tmpArr)].var(), expected_var, rtol=0.1)
+        # Uncorrected diffim exposure - variance plane is wrong (too low)
+        statObj = afwMath.makeStatistics(diffExp.getMaskedImage().getVariance(),
+                                         diffExp.getMaskedImage().getMask(), afwMath.MEANCLIP,
+                                         self.statsControl)
+        mn = statObj.getValue(afwMath.MEANCLIP)
+        self.assertNotClose(mn, expected_var, rtol=0.02)
+        print('UNCORRECTED VARIANCE:', var, mn)
 
         task = DecorrelateALKernelTask()
-        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernel)
+        mKernelPsf = measAlg.KernelPsf(mKernel)
+        decorrResult = task.run(self.im1ex, self.im2ex, diffExp, mKernelPsf)
         corrected_diffExp = decorrResult.correctedExposure
 
         # Corrected diffim - variance should be close to expected.
-        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCE,
+        # We set the tolerance a bit higher here since the simulated images have many bright stars
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage(), afwMath.VARIANCECLIP,
                                          self.statsControl)
-        var = statObj.getValue(afwMath.VARIANCE)
-        print(var, expected_var)
-        self.assertClose(var, expected_var, rtol=0.02)
+        var = statObj.getValue(afwMath.VARIANCECLIP)
+        self.assertClose(var, expected_var, rtol=0.05)
+
+        # Check statistics of variance plane in corrected diffim
+        statObj = afwMath.makeStatistics(corrected_diffExp.getMaskedImage().getVariance(),
```

Comment:
Seeing all of these makeStatistics calls, I feel like you want a couple helper functions in ImageDecorrelation.py that take MaskedImages and return the mean, variance, etc. Would clean this up a fair bit, since there's a lot of boilerplate here.